src/main/java/uk/gov/hmcts/cmc/claimstore/rules/ClaimantResponseRule.java

```diff
@@ -30,6 +59,21 @@ public void assertCanBeRequested(Claim claim, String claimantId) {
         }
     }
 
+    private void checkCourtDeterminationAndPaymentIntention(ResponseAcceptation responseAcceptation) {
+        Optional<CourtDetermination> courtDetermination = responseAcceptation.getCourtDetermination();
+        Optional<PaymentIntention> claimantPaymentIntention = responseAcceptation.getClaimantPaymentIntention();
+
+        boolean isBothEmpty = !claimantPaymentIntention.isPresent() && !courtDetermination.isPresent();
+        boolean isBothPresent = (claimantPaymentIntention.isPresent() || courtDetermination.isPresent())
+            && (claimantPaymentIntention.isPresent() && courtDetermination.isPresent());
+        if (!isBothEmpty && !isBothPresent) {
```

Comment:
This is XOR:
```
if (courtDetermination.isPresent() ^ claimantPaymentIntention.isPresent()) {
...
```

Reply:
its not though.... XOR is when one is true, the other false but false when both true or both false. both true is a valid state. i just thought more readable like this. or am i missing something? :-/